src/huggingface_hub/_commit_api.py

```diff
@@ -622,12 +647,16 @@ def _fetch_files_to_copy(
                     response = get_session().get(url, headers=headers)
                     hf_raise_for_status(response)
                     files_to_copy[(src_repo_file.path, src_revision)] = response.content
+        # 3. Ensure all operations found a corresponding file in the Hub
+        #  and track src/dest OIDs for each operation.
         for operation in operations:
             if (operation.src_path_in_repo, src_revision) not in files_to_copy:
                 raise EntryNotFoundError(
                     f"Cannot copy {operation.src_path_in_repo} at revision "
                     f"{src_revision or revision}: file is missing on repo."
                 )
+            operation._src_oid = oid_info.get((operation.src_path_in_repo, operation.src_revision))
+            operation._dest_oid = oid_info.get((operation.path_in_repo, None))
```

Comment:
Same here if comment above is correct 

```suggestion
            operation._dest_oid = oid_info.get((operation.path_in_repo, revision))
```